imports/plugins/core/accounts/server/methods/addressBookRemove.js

```diff
@@ -38,17 +38,17 @@ export default function addressBookRemove(addressId, accountUserId) {
     }
   }, { bypassCollection2: true });
 
-  // forceIndex when removing an address
-  Hooks.Events.run("afterAccountsUpdate", userId, {
-    accountId: account._id,
-    updatedFields: ["forceIndex"]
-  });
-
-  // If the address remove was successful, then return the removed address
-  if (updatedAccountResult === 1) {
-    // Pull the address from the account before it was updated and return it
-    return account.profile.addressBook.find((removedAddress) => addressId === removedAddress._id);
+  if (updatedAccountResult !== 1) {
+    throw new ReactionError("server-error", "Unable to remove address from account");
   }
 
-  throw new ReactionError("server-error", "Unable to remove address from account");
+  const updatedAccount = Accounts.findOne({ userId });
+  Promise.await(appEvents.emit("afterAccountUpdate", {
+    updatedBy: authUserId,
+    updatedAccount
+  }));
+
+  // If the address remove was successful, then return the removed address
+  // Pull the address from the account before it was updated and return it
+  return account.profile.addressBook.find((removedAddress) => addressId === removedAddress._id);
```

Comment:
Do we need to check that `updatedAccount.profile.addressBook` exists here?

Reply:
In this case, yes we should be checking because it's coming from `account` rather than `updatedAccount`. We should actually be checking much earlier in this function. I rewrote this a bit and pushed a commit.